Remove GitLFS Dependency

crvernon opened the pull request:
The GitLFS dependency for this repository has proven to be problematic for many users.  However, Xanthos still has the need to include larger files as an example directory.  To solve this issue I completed the following:

- Zipped the existing example directory and minted a DOI for it using Zenodo that establishes a supplemental data package for the Xanthos >= 2.2.0 releases,
- Built an `InstallSupplement` class that fetches and unzips the Zenodo example data package into the Xanthos root directory,
- Call the `InstallSupplement` class after `setup` has completed in `setup.py`,
- Removed and replaced any GitLFS archived files that existed within the project,
- Adjusted the paths in the `test` configuration files to recognize the working directory.

All tests have passed as executed via Travis.

Resolves #38

crvernon committed:
restructuring for to extract example dir

crvernon committed:
building class for url install of data

crvernon committed:
revise to remove GitLFS procedure

crvernon committed:
finalize install supplement fetch

crvernon committed:
add install_supplements call to setup

crvernon committed:
add install_supplements call to setup

crvernon committed:
remove GitLFS attributes file

crvernon committed:
adding setup.py install

crvernon committed:
replacing keyword var name

crvernon committed:
call error fix to fetch

crvernon committed:
add abspath call to root dir

crvernon committed:
correct paths

crvernon committed:
raise paths to class level

crvernon committed:
fix typo

crvernon committed:
fix root path

crvernon committed:
bring paths to class level

crvernon committed:
bring paths to class level

crvernon committed:
update user on supplement install

crvernon committed:
add class token

crvernon committed:
good grief - remove token

crvernon committed:
dump call to setup

crvernon committed:
dump lfs on this

crvernon committed:
make travis run setup

crvernon committed:
moving InstallSupplement to end

crvernon committed:
change to pip install .; remove 2.7

crvernon committed:
install reqs

crvernon committed:
spacing

crvernon committed:
adding requests mod

crvernon committed:
adjusting import

crvernon committed:
set to call setup.py

crvernon committed:
adding report message for unzip

crvernon committed:
adjust example dir path

crvernon committed:
update path in config for example dir

crvernon committed:
break link to lfs for docs

crvernon committed:
add in docs

crvernon commented:
I may need to un-init GitLFS to make the dependency actually vanish; though the heavy-lifting in terms of LFS file installation is done.  I'll check into this.

rplzzz commented on setup.py:
```diff
@@ -44,3 +43,9 @@ def get_requirements():
+
+# install supplemental and example data from Zenodo
+print("Installing supplemental data from Zenodo...")
+InstallSupplement().fetch_zenodo()
```
If you don't mind, I'd like to add a newline to the end of this file.
```suggestion
InstallSupplement().fetch_zenodo()

```

crvernon replied:
Good for me. I meant to as well.

rplzzz commented on xanthos/install_supplement.py:
```diff
@@ -0,0 +1,36 @@
+    def __init__(self):
+
+        # full path to the Xanthos root directory where the example dir will be stored
+        self.root_dir = os.path.split(os.path.dirname(__file__))[0]
```
Will this work if the package is installed as an egg or wheel file?

If so, would it not be better to use `pkg_resources` to discover the package directory?  I guess technically `__file__` does the same thing, but using it always makes me feel like I'm doing something shady.

crvernon replied:
You are correct.  I was doing something shady!  I will fix this up.

rplzzz replied:
Ok.  I'll take another look tomorrow.

rplzzz replied:
@crvernon Does this fix the problem that we identified?  It looks like we are still trying to write the files into the directory with the python package.  Are we certain this will work in all cases?

crvernon replied:
@rplzzz I haven't made the fix yet.

rplzzz replied:
My bad.  Github popped it up on my dashboard in a way that made it seem like you were waiting on me.

rplzzz reviewed (commented):
See question in `install_supplement.py`.  Is it kosher to add files to the distribution after it has been unpacked?

rplzzz committed:
Update setup.py

add new line at end of file

Co-Authored-By: crvernon <chrisrvernon@gmail.com>

rplzzz commented:
@crvernon How are we doing on this task?  I haven't seen an update in a while.

crvernon committed:
alter install to get current version

crvernon committed:
remove install supplement from setup

crvernon committed:
add include manifest pkgs in setup

crvernon committed:
build internal data load

crvernon committed:
remove reference loading from ini

crvernon committed:
adding manifest

crvernon committed:
adding package data

crvernon committed:
add test data to manifest

crvernon committed:
setup for new non-lfs config

crvernon committed:
update to fetch via pkg_resources

crvernon committed:
update to fetch via pkg_resources

crvernon committed:
fix formatting

crvernon committed:
removing old data

crvernon committed:
removing test data from manifest

crvernon committed:
fix file paths

crvernon committed:
adding example install script for travis-ci

crvernon committed:
update data install

crvernon committed:
fix paths

crvernon committed:
remove install data call

crvernon committed:
fix config path

crvernon committed:
remove os import

crvernon committed:
remove lfs

crvernon committed:
raise class to main

crvernon commented:
Alright @rplzzz we are ready to roll with a review on this one.  

- I made the reference data part of the package which is called accordingly
- replaced all `__file__` calls with proper package lingo using `pkg_resources`
- dumped the auto-install of the example data on setup
- removed all `GitLFS` pointers and objects (see https://github.com/git-lfs/git-lfs/issues/3026)
- set up Travis-CI to install the example data for tests in the travis config outside
- fixed all paths in all tests, etc.

The protocol for installing the example data for Xanthos is now conducted by executing the following after install:
```python
from xanthos import InstallSupplement
InstallSupplement("<path to the directory you wish to store the example data in>")```

Once the the current checks have completed - they will pass - you can review.  After I get the OK from you, I will add this install protocol to the README (BTW, we need a readthedocs.io page) and mint the new example data for usage with the most recent version.

rplzzz commented on xanthos/install_supplement.py:
```diff
@@ -0,0 +1,73 @@
+    args = parser.parse_args()
+
+    zen = InstallSupplement(args.example_data_directory)
+    del zen
```
Is this `del` statement needed?  The object should be gc'd when the script finishes anyhow, right?  Or is there something I'm missing?

rplzzz reviewed (approved):
Everything looks good.  Can you resolve the conflicts with master?  It looks like they're all binary files.

crvernon committed:
remove existing test structure

crvernon committed:
add new tests dir

crvernon committed:
updating tests and data loading protocol

crvernon committed:
working inheritance

crvernon committed:
liberate pm data loader; set test

crvernon committed:
creating arg passing for abcd data loader

crvernon committed:
working on test validation for pm pet
